src/libbson/CMakeLists.txt

```diff
@@ -267,6 +267,10 @@ if (ENABLE_STATIC MATCHES "ON|AUTO")
    target_include_directories (bson_static INTERFACE $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/src> $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}/src>)
    set_target_properties (bson_static PROPERTIES VERSION 0.0.0)
    set_target_properties (bson_static PROPERTIES OUTPUT_NAME "${BSON_OUTPUT_BASENAME}-static-${BSON_API_VERSION}")
+   if (ENABLE_WINDOWS_STATIC_RUNTIME)
```

Comment:
Can this be changed to a cached option? Similar to options here: https://github.com/mongodb/mongo-c-driver/blob/1.15.0/CMakeLists.txt/#L57

A cached option will show up after configuring with cmake, ending up in sort of a help menu. Users can get a list of cached options along with the help text by running `cmake -LH` or using  cmake GUI or ccmake.

Arguably, the same should be done for https://github.com/mongodb/libmongocrypt/pull/78, but I think it's much less likely anyone outside of our drivers team is using libmongocrypt directly.

Reply:
I'm not familiar with CMake, but should this option only exist on Windows platforms?

Coming from an Autotools perspective, I am imagining that this behaves similar to a configure option, where we'd simply prefer to have it omitted entirely from the list of supported options if we know it's not relevant to the platform. If that's not the case for CMake, please correct me.

Reply:
Sorry, some discussion/investigation has happened outside of this PR and I did not follow-up. This PR can hopefully be closed without merging. I was able to pass `/MT` through the relwithdebinfo Windows compile using `-DCMAKE_C_FLAGS_RELWITHDEBINFO`. @mbroadst if you can verify the changes on this branch work (https://github.com/kevinAlbs/libmongocrypt/tree/proposed-fix), shall we close this PR?

Though I do agree it'd probably make sense to have this as a Windows only option.